botocore/signers.py

```diff
@@ -123,6 +113,36 @@ def sign(self, operation_name, request):
                                             signature_version)
             signer.add_auth(request=request)
 
+    def choose_signer(self, operation_name, suffix=''):
+        """
+        Allow overriding the signature version via the choose-signer event.
+        A value of `botocore.UNSIGNED` means no signing will be performed.
```

Comment:
This docstring doesn't seem right.  We're not really "overriding" signature versions.  We're simply giving it an operation, and asking what signature version the request signer would use to sign the request.